kms/keysource.go

```diff
@@ -140,12 +140,22 @@ func (key MasterKey) createStsSession(config aws.Config, sess *session.Session)
 }
 
 func (key MasterKey) createSession() (*session.Session, error) {
-	re := regexp.MustCompile(`^arn:aws:kms:(.+):([0-9]+):key/(.+)$`)
+	// nb. aws/sdk-go/aws/endpoints
+	validPartition := map[string]int {
+		"aws": 1,
+		"aws-cn": 2,
+		"aws-us-gov": 3,
+	}
+	re := regexp.MustCompile(`^arn:(aws[\w-]*):kms:(.+):([0-9]+):key/(.+)$`)
 	matches := re.FindStringSubmatch(key.Arn)
 	if matches == nil {
 		return nil, fmt.Errorf("No valid ARN found in %q", key.Arn)
 	}
-	config := aws.Config{Region: aws.String(matches[1])}
+	_, exists := validPartition[matches[1]]
```

Comment:
I don't think this check is needed. We don't currently check that the region is valid, so I don't see why we should check whether the partition is valid or not. In my opinion we should just expand the regex to

```
re := regexp.MustCompile(`^arn:aws[\w-]*:kms:(.+):([0-9]+):key/(.+)$`)
```